test/validate-schemas.spec.ts

```diff
@@ -0,0 +1,64 @@
+import fs from 'fs-extra';
+import upath from 'upath';
+import { Json } from '../lib/util/schema-utils';
+import { capitalize } from '../tools/docs/utils';
+import * as Schemas from '../tools/schemas/schema';
+
+const dataFileDir = 'lib/data';
+const schemaDir = 'tools/schemas';
+const schemasAndJsonFiles: {
+  schemaName: keyof typeof Schemas;
+  dataFileName: string;
+}[] = [];
+
+describe('validate-schemas', () => {
+  it('validate json files in lib/data against their schemas', async () => {
+    const schemaFiles = (await fs.readdir(schemaDir)).filter(
+      (file) => upath.extname(file) === '.json',
+    );
+
+    for (const schemaFile of schemaFiles) {
+      const correspondingDatFileName = schemaFile.replace('-schema', '');
+      const schemaName = `${schemaFile
+        .replace('.json', '')
+        .split('-')
+        .map(capitalize)
+        .join('')}` as keyof typeof Schemas;
+      schemasAndJsonFiles.push({
+        schemaName,
+        dataFileName: correspondingDatFileName,
+      });
+    }
+
+    const settledPromises: any = await Promise.allSettled(
```

Comment:
```suggestion
    const settledPromises = await Promise.allSettled(
```
why any?

Reply:
TS isn't detecting the correct type for this from the respective `.d.ts` file.

The `.d.ts` file defines the return type for `allSettled` as `PromiseFulfilledType | PromiseRejectedType` but TS only infers `PromiseFulfilledType`, due to which I get TS errors when I use the `reason` field, which is defined in `PromiseRejectedType`

Reply:
remove the reason, status compare should be enough

Reply:
The reason holds the invalid json OR invalid schema message so its relevant while debugging failed tests

Reply:
I did a new commit after only removing the `any` type assertion and not TS errors. Maybe it was a one time bug